File: dagger-scope/sample/src/main/java/ru/surfstudio/android/custom_scope_sample/ui/base/dagger/screen/CustomScreenModule.java
package ru.surfstudio.android.custom_scope_sample.ui.base.dagger.screen;


import dagger.Module;
import dagger.Provides;
import ru.surfstudio.android.core.ui.navigation.Route;
import ru.surfstudio.android.dagger.scope.PerScreen;

/**
 * базовый класс для дополнительного модуля конкретного экрана (обычно используется для поставки
 * параметров, переданных на экран при старте в презентер, в этом случае параметры передаются обурнутые в Route)
 *
 * @param <R>
 */
@Module
public abstract class CustomScreenModule<R extends Route> {

    private R route;

    public CustomScreenModule(R route) {
        this.route = route;
    }

    @Provides
    @PerScreen
    public R provideRoute() {
        return route;
    }
}

File: template/base_feature/src/main/java/ru/surfstudio/standard/ui/screen_modules/CustomScreenModule.kt
package ru.surfstudio.standard.ui.screen_modules

import dagger.Module
import dagger.Provides
import ru.surfstudio.android.dagger.scope.PerScreen
import ru.surfstudio.android.navigation.route.Route

/**
 * базовый класс для дополнительного модуля конкретного экрана (обычно используется для поставки
 * параметров, переданных на экран при старте в презентер, в этом случае параметры передаются, обернутые в Route)
 *
 * @param <R>
</R> */
@Module
abstract class CustomScreenModule<out R : Route>(private val route: R) {

    @Provides
    @PerScreen
    fun provideRoute(): R {
        return route
    }
}